python-sdk/src/astro/databases/aws/redshift.py

```diff
@@ -104,6 +104,157 @@ def load_pandas_dataframe_to_table(
+        :param target_conflict_columns: list of columns where we expect to have a conflict while combining
+        """
+        conflict_columns_str = ",".join(target_conflict_columns)
+        insert_statement = (
+            f"INSERT INTO {stage_table_name}({target_table_all_columns_string}) "
+            f"SELECT {source_table_all_columns_string} FROM {source_table_name} "
+            f"WHERE ({conflict_columns_str}) "
+            f"NOT IN (SELECT {conflict_columns_str} FROM {stage_table_name})"
+        )
+
+        conflict_statements = None
+        if if_conflicts == "ignore":
+            conflict_statements = [insert_statement]
+        elif if_conflicts == "update":
+            conflict_column = target_conflict_columns[0]
+            update_statement = (
+                f"UPDATE {stage_table_name} "
+                f"SET {source_to_target_map_target_columns[0]}="
+                f"{source_table_name}.{source_to_target_map_source_columns[0]}"
+            )
+            for (source_column, target_column) in zip(
+                source_to_target_map_source_columns[1:],
+                source_to_target_map_target_columns[1:],
+            ):
+                update_statement += (
+                    f", {target_column}={source_table_name}.{source_column} "
+                )
+            update_statement += (
+                f"FROM {source_table_name} "
+                f"WHERE {stage_table_name}.{conflict_column}={source_table_name}.{conflict_column} "
+            )
+            for conflict_col in target_conflict_columns[1:]:
+                update_statement += f"AND {stage_table_name}.{conflict_col}={source_table_name}.{conflict_col} "
+            conflict_statements = [update_statement, insert_statement]
+
+        return conflict_statements
+
+    def merge_table(
+        self,
+        source_table: Table,
+        target_table: Table,
+        source_to_target_columns_map: Dict[str, str],
+        target_conflict_columns: List[str],
+        if_conflicts: MergeConflictStrategy = "exception",
+    ) -> None:
+        """
+        Merge the source table rows into a destination table.
+        The argument `if_conflicts` allows the user to define how to handle conflicts.
+
+        :param source_table: Contains the rows to be merged to the target_table
+        :param target_table: Contains the destination table in which the rows will be merged
+        :param source_to_target_columns_map: Dict of target_table columns names to source_table columns names
+        :param target_conflict_columns: List of cols where we expect to have a conflict while combining
+        :param if_conflicts: The strategy to be applied if there are conflicts.
+        """
+        source_table_name = self.get_table_qualified_name(source_table)
+        target_table_name = self.get_table_qualified_name(target_table)
+        stage_table_name = self.get_table_qualified_name(Table())
```

Comment:
@pankajkoti I think it would be a good idea to create this table in the temp schema. right now we are creating this in the default database scheme. WDYT?

Reply:
Redshift doe not accept schema option for temp tables. As per the docs here: https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html

TEMPORARY | TEMP
Keyword that creates a temporary table that is visible only within the current session. The table is automatically dropped at the end of the session in which it is created. The temporary table can have the same name as a permanent table. The temporary table is created in a separate, session-specific schema. (You can't specify a name for this schema.) This temporary schema becomes the first schema in the search path, so the temporary table will take precedence over the permanent table unless you qualify the table name with the schema name to access the permanent table. For more information about schemas and precedence, see [search_path](https://docs.aws.amazon.com/redshift/latest/dg/r_search_path.html).